packages/grafana-ui/src/components/VizLegend/SeriesIcon.tsx

```diff
@@ -29,28 +31,47 @@ export const SeriesIcon = React.memo(
       cssColor = color!;
     }
 
-    const styles: CSSProperties = {
-      background: cssColor,
-      width: '14px',
-      height: '4px',
-      borderRadius: theme.shape.radius.pill,
-      display: 'inline-block',
-      marginRight: '8px',
-    };
+    const lineStyleStyle =
+      lineStyle?.fill === 'dot' ? styles.dot : lineStyle?.fill === 'dash' ? styles.dash : styles.solid;
+
+    const colorStyle: CSSProperties =
+      lineStyle?.fill === 'dot' || lineStyle?.fill === 'dash'
+        ? {
+            borderTopColor: cssColor,
+          }
+        : {
+            backgroundColor: cssColor,
+          };
```

Comment:
we can probably use a fat border for all cases (even solid)?

Reply:
done!